pkg/pillar/hypervisor/kvm.go

```diff
@@ -459,6 +454,57 @@ func (ctx kvmContext) Task(status *types.DomainStatus) types.Task {
 	}
 }
 
+// The whole domain is considered as a container-base if one of the disks
+// has the 'CONTAINER' format.
+func isContainer(config types.DomainConfig) bool {
```

Comment:
Based on the comment for GetOCIConfigDir() that is the way to determine whether a task should run as a container (in a VM).
It makes sense using that to create an IsContainer member function in DomainConfig.

In fact, the string returned by GetOCIConfigDir() isn't used, so you can just change that functions name to IsContainer() and its return type to be boolean.